### ISSUE TITLE
how to document custom IFC properties

### ISSUE THREAD

**Issue by mathib** [buildingSMART/IFC#23]
Hi all,
I know that a lot of things are changing regarding the definition of products and properties, but I would like to know what is the current "standard" way to define **custom** properties for IFC building objects in an interoperable manner. I know it is possible to define custom properties at will in IFC files, but I'm looking for methods to document them properly. Is there a standard or bSI reference that indicates how this should be done? Should they follow the PSD-XML structure, e.g. as done for predefined properties and PSets: https://standards.buildingsmart.org/IFC/DEV/IFC4_2/FINAL/HTML/psd/? If yes, is there an XSD or any documentation of it?

**Comment by TLiebich** [comment]
yes, you can document it using the psd xml format. Alternatively you can use (since IFC4) also a subset of the IFC4 specification, namely the `IfcPropertyTemplate `and `IfcPropertyTemplate `to express the definition of a property set. See here for an example:

- https://standards.buildingsmart.org/IFC/RELEASE/IFC4/ADD2_TC1/HTML/link/listing-ifc4_add2.htm

the definition of the XSD for the property set XML format exists, it was part of the previous buildingsmart-tech.org site. @jwouellette - lets have a look if we can provide it again.

**Comment by mathib** [comment]
Thanks. I've had a hard time finding just the existing PSDs and could indeed not find the XSD! Would be good to have it on a stable location. Maybe the XSD and PSDs can also be provided in this Github repo?

I saw that this repo already has [properties in XML format](https://github.com/buildingSMART/IFC/tree/master/Properties), but they seem to follow a different (non standard?) XML structure. What is the difference?

Is PSD in XML format actually defined in any of the ISO standards (e.g. ISO 16739-1:2018 or others)?

**Comment by jmirtsch** [comment]
I believe PSD is a bespoke data serialization.  They are also published in accordance with IFC which is an ISO standard.  You can find the ifcxml file listed on this page, https://standards.buildingsmart.org/IFC/DEV/IFC4_3/RC1/HTML/link/listing-ifc4x3_rc1.htm

I would like to see an official way of nesting translations within the property templates, this would allow quite a neat version control of this serialization of the properties.
